### PR TITLE
JKRDecomp OK

### PR DESCRIPTION
**Pull request by Julgodis**

### CODE REVIEW

File: libs/JSystem/JKernel/JKRThread.cpp [hunk 1]
```diff
@@ -51,21 +51,21 @@ void JKRThread::setCommon_mesgQueue(JKRHeap* heap, int message_count) {
     this->mMessageCount = message_count;
     this->mMessages = (OSMessage*)JKRHeap::alloc(this->mMessageCount * sizeof(OSMessage), 0, heap);
 
-    OSInitMessageQueue(&this->mQueue, this->mMessages, this->mMessageCount);
+    OSInitMessageQueue(&this->mMessageQueue, this->mMessages, this->mMessageCount);
     lbl_8043428C.append(&this->mThreadListLink);
 
-    this->field_0x74 = (JKRHeap*)NULL;
-    this->field_0x78 = (JKRHeap*)NULL;
+    this->mCurrentHeap = NULL;
```
Comment (notyourav) [inline libs/JSystem/JKernel/JKRThread.cpp:57]:
```suggestion
    mCurrentHeap = NULL;
```

File: libs/JSystem/JKernel/JKRThread.cpp [hunk 2]
```diff
@@ -51,21 +51,21 @@ void JKRThread::setCommon_mesgQueue(JKRHeap* heap, int message_count) {
...
+    this->mCurrentHeapError = NULL;
 }
 
 // #include "JSystem/JKernel/JKRThread/asm/func_802D18A4.s"
 void JKRThread::setCommon_heapSpecified(JKRHeap* heap, u32 stack_size, int param_3) {
     this->mHeap = heap;
     this->mStackSize = stack_size & 0xffffffe0;
-    this->mStackPtr = JKRHeap::alloc(this->mStackSize, 0x20, this->mHeap);
+    this->mStackBase = JKRHeap::alloc(this->mStackSize, 0x20, this->mHeap);
```
Comment (notyourav) [inline libs/JSystem/JKernel/JKRThread.cpp:65]: Does this work?
```suggestion
    mStackBase = JKRAllocFromHeap(0x20, mHeap, mStackSize, param_3); // JKRHeap* param_3
```

```cpp
// m_Do_ext.h inline
inline void JKRAllocFromHeap(int alignment, JKRHeap* heap, u32 size, JKRHeap* parent) {
    heap->alloc(size, alignment, parent);
}
```

**Review by notyourav — CHANGES_REQUESTED** [review]

File: libs/JSystem/JKernel/JKRThread.cpp (hunk 2 above)
Comment (Julgodis) [reply libs/JSystem/JKernel/JKRThread.cpp]: Yeah, the debug version uses JKRAllocFromHeap. 
I changed it from JKRHeap::alloc to use JKRAllocFromHeap (added the function to JKRHeap.h).

(FYI, that is not the correct function signature for JKRAllocFromHeap)

Comment (notyourav) [reply libs/JSystem/JKernel/JKRThread.cpp]: Yeah, I was having some trouble figuring it out :P

**Review by notyourav — APPROVED** [review]

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**